pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/ForwardIndexHandler.java

```diff
@@ -302,8 +302,11 @@ Map<String, List<Operation>> computeOperations(SegmentDirectory.Reader segmentRe
         }
 
         // Note that RAW columns cannot be sorted.
+        // TODO: revisit the the possibility of converting noDictionary -> Dictionary when the column is sorted.
         ColumnMetadata existingColMetadata = _segmentDirectory.getSegmentMetadata().getColumnMetadataFor(column);
-        Preconditions.checkState(!existingColMetadata.isSorted(), "Raw column=" + column + " cannot be sorted.");
+        if (existingColMetadata.isSorted()) {
```

Comment:
Can we please add a LOGGER.warn here with the column name?

Reply:
added!